sympy/sets/sets.py

```diff
@@ -794,6 +794,12 @@ def as_relational(self, *symbols):
                 'number of symbols must match the number of sets')
         return And(*[s.as_relational(i) for s, i in zip(self.sets, symbols)])
 
+    def _eval_evalf(self, prec):
+        try:
+            return self.func(*[arg._evalf(prec) for arg in self.args])
+        except (TypeError, ValueError, NotImplementedError):
+            raise TypeError("Not all sets are evalf-able")
```

Comment:
I would like to limit exception catching as much as possible because it often ends up masking bugs. Is there a need to catch all of these?

Reply:
I don't think it is nessesary. I'll remove them.